src/player.ts

```diff
@@ -1,32 +1,79 @@
+export enum PlayerState {
+    ON_GROUND,
+    IN_AIR,
+    LEFT_SLIDE,
+    RIGHT_SLIDE
+}
 export class Player extends Phaser.Physics.Arcade.Sprite {
     jumpCount: number
     body: Phaser.Physics.Arcade.Body
-    inAir: boolean
+    state: PlayerState
+    movingLeft: boolean
+    movingRight: boolean
+    //constants. modify these to tweak 
+    static GRAVITY: number = 50;
+    static X_INC: number = 60;
+    static TOP_SPEED: number = 600;
 
-    constructor(scene: Phaser.Scene,x: number, y: number,  image: string ) {
+    constructor(scene: Phaser.Scene, x: number, y: number, image: string) {
         super(scene, x, y, image)
         this.jumpCount = 2;
         this.body = new Phaser.Physics.Arcade.Body(scene.physics.world, this)
-        this.inAir = false;
+        this.state = PlayerState.IN_AIR;
+        this.movingLeft = false;
+        this.movingRight = false;
     }
 
     jump() {
```

Comment:
This should be handled as part of the player sprite's state machine update; the player sprite should be checking what the inputs are during the state machine update, enter an `IN_AIR` state, and set the velocity.